images/router/haproxy/conf/haproxy-config.template

```diff
@@ -252,7 +252,13 @@ backend be_tcp_{{$cfgIdx}}
 {{ if ne (env "ROUTER_SYSLOG_ADDRESS" "") ""}}
   option tcplog
 {{ end }}
+    {{ with $balanceAlgo := index $cfg.Annotations "router.openshift.io/haproxy.balance" }}
+      {{ with $matchValue := (matchValues $balanceAlgo "roundrobin" "leastconn" ) }}
+  balance {{ $balanceAlgo }}
```

Comment:
I'm just template ignorant. What value does https://github.com/openshift/origin/pull/9710/files#diff-dc1a3cefa4a5177a9bfd475c81220cf1R256 bring? Does it cause the entire template processing to fail if the match fails?